docs/branches.md

```diff
@@ -41,6 +40,48 @@ branch `main`. If the local target had also moved compared to `main@origin`
 merged. If one is ahead of the other, then that target will be the new target.
 Otherwise, the local branch will be conflicted (see next section for details).
 
+As of December 2023 Jujutsu tracks and fetches all branches by default, which 
+is confusing users coming from Git and Mercurial. To smoothen the transition 
+branch tracking was introduced. 
+
+### Tracking a branch
+
+To track a branch permanently use `jj branch track <name-of-branch>`. It will 
+now be imported as a local branch until you untrack it or it is deleted on the 
+remote. 
+
+Example:
+
+```sh
+$ # List all available branches, as we want our colleague's branch.
+$ jj git branch list --all
+$ # Find the branch.
+$ jj git branch track <name>
```

Comment:
nit: I misinterpreted that `jj git branch track` is the command to "Find the branch".

Maybe "Find the branch" can be moved to the first paragraph?

Reply:
I've added ellided content ( `[...]`) and a preciser comment above each command.